### PR TITLE
implement search by name and improve event filter

### PR DESCRIPTION
**Pull request by Rodgers-M**
**What does this PR do?**

- Implement searching events by name and improve event filter

**Description of tasks to be completed**

- write tests for the search route
- implement search using the q parameter
- improve event filter by category and location

**How can this be tested manually?**
Pull the latest changes and test the `events/search` route

**Relevant tracker stories**

- 154496559

### CODE REVIEW

File: app/events/views.py [hunk 1]
```diff
@@ -58,16 +65,16 @@ def create():
...
 
 @events.route('/all')
-@events.route('/all/<int:page>')
-def get_all(page=1):
+def get_all():
 	"""fetch all events available"""
 	#fetch the first 15 events based on event date
-	result = Events.query.filter(cast(Events.event_date, Date) >=  date.today()).paginate(page, per_page=15, error_out=False)
+	page = request.args.get("page", default=1, type=int)
+	per_page = request.args.get("per_page", default=15, type=int)
```
Comment (meshnesh) [inline app/events/views.py]: good implementation of pagination and page limiter dynamically

**Review by meshnesh — COMMENTED** [review]
* Why is the route different for searching of events and filtering, they should work on the same route.
* Good implementation of test for search and filter
* In the application can you filter an event by category and filter at the same time?

File: app/events/views.py (hunk 1 above)
Comment (Rodgers-M) [reply app/events/views.py]: no, you can only filter by one attribute at a time. though it also makes sense to use both.

Comment (meshnesh) [reply app/events/views.py]: Yes it would be advisable to have both.

Comment (Rodgers-M) [reply app/events/views.py]: just implemented that feature, check it out

**Pull request merged**